cockroachdb/replicator
# internal/target/schemawatch: added in configuration wiring for the RefreshDelay option

Previously, the RefreshDelay was set to a constant value of 1 minute, which means that customers did not have a way to opt out or update this behavior so it happens less or more frequently. This change introduces the wiring from each frontend to the schemawatch package to control this schema refresh timing.

Resolves: #1047
Release Note: Added in the ability to configure the schema watcher refresh delay via a CLI flag.

**TODO**
- [x] Fix all broken tests: note that all tests that broke are because of: `Error: The operation was canceled.`, which indicates there was a timeout for the test, which matches what we see
- [x] Wiki updates with the latest new flag here

**Manual Testing**
Put log lines right before the schema watch go routine kicks off and checks on an interval and see that it takes the value we pass in.
```
go run . start ..... --schemaRefresh '2m'
...
time="Oct 18 17:05:18" level=info msg="Replicator starting" -buildmode=exe -compiler=gc -gcflags="all=-N -l" CGO_CFLAGS="-O0 -g" CGO_CPPFLAGS= CGO_CXXFLAGS= CGO_ENABLED=1 CGO_LDFLAGS= GOARCH=arm64 GOARM64=v8.0 GOOS=darwin
time="Oct 18 17:05:18" level=info msg="authentication disabled, any caller may write to the target database"
time="Oct 18 17:05:18" level=info msg="Server listening" address="[::]:30004"
time="Oct 18 17:05:18" level=info msg="Version 8.0.39."
time="Oct 18 17:05:18" level=debug msg="Mode REAL_AS_FLOAT,PIPES_AS_CONCAT,ANSI_QUOTES,IGNORE_SPACE,ONLY_FULL_GROUP_BY,ANSI. Ssl_cipher "
2m0s
delay is:  2m0s
```


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/replicator/1048)
